flink-clients/src/test/java/org/apache/flink/client/CliFrontendStopTest.java

```
@@ -0,0 +1,157 @@
+	public static void teardown() {
+		JavaTestKit.shutdownActorSystem(actorSystem);
+		actorSystem = null;
+	}
+
+	@BeforeClass
+	public static void init() {
+		CliFrontendTestUtils.pipeSystemOutToNull();
+		CliFrontendTestUtils.clearGlobalConfiguration();
+	}
+
+	@Test
+	public void testStop() {
+		try {
+			// test unrecognized option
+			{
+				String[] parameters = { "-v", "-l" };
+				CliFrontend testFrontend = new CliFrontend(CliFrontendTestUtils.getConfigDir());
+				int retCode = testFrontend.stop(parameters);
+				assertTrue(retCode != 0);
+			}
+
+			// test missing job id
+			{
+				String[] parameters = {};
+				CliFrontend testFrontend = new CliFrontend(CliFrontendTestUtils.getConfigDir());
+				int retCode = testFrontend.stop(parameters);
+				assertTrue(retCode != 0);
+			}
+
+			// test stop properly
+			{
+				JobID jid = new JobID();
+				String jidString = jid.toString();
+
+				final Option<UUID> leaderSessionID = Option.<UUID>apply(UUID.randomUUID());
+				final ActorRef jm = actorSystem.actorOf(Props.create(CliJobManager.class, jid));
+
+				final ActorGateway gateway = new AkkaActorGateway(jm, leaderSessionID);
+
+				String[] parameters = { jidString };
+				StopTestCliFrontend testFrontend = new StopTestCliFrontend(gateway);
+
+				int retCode = testFrontend.stop(parameters);
+				assertTrue(retCode == 0);
+			}
+
+			// test stop properly
+			{
+				JobID jid1 = new JobID();
+				JobID jid2 = new JobID();
+
+				final Option<UUID> leaderSessionID = Option.<UUID>apply(UUID.randomUUID());
+				final ActorRef jm = actorSystem.actorOf(Props.create(CliJobManager.class, jid1));
+
+				final ActorGateway gateway = new AkkaActorGateway(jm, leaderSessionID);
+
+				String[] parameters = { jid2.toString() };
+				StopTestCliFrontend testFrontend = new StopTestCliFrontend(gateway);
+
+				assertTrue(testFrontend.stop(parameters) != 0);
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			fail("Program caused an exception: " + e.getMessage());
```

Why do you print the stack trace and let the test fail with an `AssertionError` instead of the original exception?